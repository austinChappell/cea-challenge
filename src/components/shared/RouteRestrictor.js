import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

const propTypes = {
  accessToken: PropTypes.string,
};

const defaultProps = {
  accessToken: null,
};

const RouteRestrictor = (props) => {
  const { accessToken } = props;
  const development = process.env.REACT_APP_ENV === 'development';
  const authorized = accessToken !== null || development;
  const content = authorized ? null : <Redirect to="/" />;
  return content;
};

const mapStateToProps = state => ({
  accessToken: state.generalReducer.accessToken,
});

RouteRestrictor.propTypes = propTypes;
RouteRestrictor.defaultProps = defaultProps;

export default connect(mapStateToProps)(RouteRestrictor);
