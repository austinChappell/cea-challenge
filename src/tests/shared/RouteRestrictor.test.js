import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import RouteRestrictor from '../../components/shared/RouteRestrictor';

const initialState = {
  generalReducer: {
    accessToken: 'fakeToken',
  },
};

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('The RouteRestrictor Component', () => {
  it('should render correctly', () => {
    const options = {
      context: {
        store: mockStore(initialState),
      },
    };

    const wrapper = shallow(<RouteRestrictor />, options);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
