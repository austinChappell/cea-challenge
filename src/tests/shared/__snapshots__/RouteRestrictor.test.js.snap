// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RouteRestrictor Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RouteRestrictor
    accessToken="fakeToken"
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
